src/lib/logger.ts

```diff
@@ -130,15 +131,22 @@ export enum LogEventName {
   TEACHER_NETWORK_COLLAPSE_DOCUMENT_SECTION,
 }
 
-// This is the form the log events take  
+// This is the form the log events take
 export interface SimpleToolLogEvent {
   path?: string;
   args?: Array<any>;
 }
 
-type LoggableToolChangeEvent = Optional<JXGChange, "operation"> |
+export interface DataflowProgramChange extends Record<string,any>{
+  targetType: string // | Node | Connection | Control ?
```

Comment:
Need a trailing `,` or `;` here?

Also not sure if you were having trouble with `| Node | Connection | Control ?` and gave up. If so, would adding quotes around the options work?

Reply:
I think this is a vestige from a time when I thought I'd be able to always pass a rete object as a target.  And yes, it needed a comma!